plugins/module_utils/statuscake.py

```diff
@@ -37,33 +37,37 @@ def fetch(self):
                     logger.debug(f"Fetched data: {test}")
                     return test
 
-    def create(self):
-        if "test_type" not in self.data:
-            self.data["test_type"] = "HTTP"
-        if "check_rate" not in self.data:
-            self.data["check_rate"] = 300
-        self._request("post", self.url, data=self.data)
+    def _create(self):
+        if self.data['state'] == 'present':
+            if "test_type" not in self.data:
+                self.data["test_type"] = "HTTP"
+            if "check_rate" not in self.data:
+                self.data["check_rate"] = 300
+            self._request("post", self.url, data=self.data)
 
-    def update(self, item_id):
+    def _update(self, item_id):
         if "test_type" not in self.data:
             self.data["test_type"] = "HTTP"
         if "check_rate" not in self.data:
             self.data["check_rate"] = 300
         self._request("put", f"{self.url}/{item_id}", data=self.data)
     
-    # def delete(self):
-    #     fetch_data = self.fetch()
-    #     self._request("delete", f"{self.url}/{fetch_data['id']}", data=self.data)
+    def delete(self):
```

Comment:
I'd suggest matching the API signature of `update`. Otherwise, a user of the class asks the question: why is `item_id` passed to `update()`, but not passed to `delete()`?

Reply:
We decided to work on this together. The script will resemble Django in that we will use the id as the main identifier of a test.